theano/gpuarray/opt.py

```diff
@@ -1716,6 +1830,70 @@ def local_abstractconv_gradweights_gemm(node):
     return [rval]
 
 
+@local_optimizer([AbstractConv2d_gradWeights])
+def local_abstractconv_gemm_gradweights_alt(node):
```

Comment:
there seem to be much code duplicate between the 2d and 3d case and cudnn and gemm alternative.

Do you think it is easy to refactor?